### PR TITLE
Figma2Studio part3

### PR DESCRIPTION
**Pull request by vaiwa**

### CODE REVIEW

File: packages/octopus-fig/test/integration/figma-parser/assets/79r1c5joN7rUmT2StGPPGA/expected/octopus-1-2.json [hunk 1]
```diff
@@ -34,6 +34,18 @@
         "dimensions": {
           "width": 375,
           "height": 250
+        },
+        "absoluteBoundingBox": {
+          "x": -5.684341886080802e-14,
```
Comment (cerosnikita) [inline packages/octopus-fig/test/integration/figma-parser/assets/79r1c5joN7rUmT2StGPPGA/expected/octopus-1-2.json:39]: I'd round it to `0`.

File: packages/octopus-fig/src/entities/octopus/octopus-layer-mask-group.ts [hunk 2]
```diff
@@ -217,7 +217,9 @@ export class OctopusLayerMaskGroup {
 
   get meta(): Octopus['LayerMeta'] | undefined {
     const dimensions = this.dimensions
-    return { dimensions, isArtboard: true }
+    const absoluteBoundingBox = this.sourceLayer.absoluteBoundingBox
+    const absoluteRenderBounds = this.sourceLayer.absoluteRenderBounds
```
Comment (cerosnikita) [inline packages/octopus-fig/src/entities/octopus/octopus-layer-mask-group.ts:221]: As far as I understand both `absoluteBoundingBox` and `absoluteRenderBounds` could be `undefined`. If it's `undefined`, it shouldn't be part of result.

File: packages/octopus-fig/src/entities/octopus/octopus-path.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ export class OctopusPath {
       case 'RECTANGLE':
         return 'RECTANGLE'
       case 'LINE':
+      case 'VECTOR':
```
Comment (cerosnikita) [inline packages/octopus-fig/src/entities/octopus/octopus-path.ts:41]: Why `VECTOR`s are not converted to `LINE`s? 😲

File: packages/octopus-fig/src/entities/source/source-layer-common.ts [hunk 4]
```diff
@@ -151,4 +151,16 @@ export class SourceLayerCommon extends SourceEntity {
     if (this._rawValue.parent?.type) return this._rawValue.parent?.type === 'PAGE'
     return this.parent instanceof SourceArtboard
   }
+
+  get absoluteBoundingBox(): Octopus['Bounds'] | undefined {
+    const { x, y, width, height } = this._rawValue.absoluteBoundingBox ?? {}
+    if (x === undefined || y === undefined || width === undefined || height === undefined) return undefined
```
Comment (cerosnikita) [inline packages/octopus-fig/src/entities/source/source-layer-common.ts:157]: Or you can just:
```ts
if ([x, y, width, height].some(n => typeof n !== 'number')) return undefined
```

File: packages/octopus-fig/src/entities/octopus/octopus-path.ts (hunk 3 above)
Comment (vaiwa) [reply packages/octopus-fig/src/entities/octopus/octopus-path.ts]: reverted

File: packages/octopus-fig/src/entities/octopus/octopus-layer-mask-group.ts (hunk 2 above)
Comment (vaiwa) [reply packages/octopus-fig/src/entities/octopus/octopus-layer-mask-group.ts]: absoluteBounds removed

File: packages/octopus-fig/test/integration/figma-parser/assets/79r1c5joN7rUmT2StGPPGA/expected/octopus-1-2.json (hunk 1 above)
Comment (vaiwa) [reply packages/octopus-fig/test/integration/figma-parser/assets/79r1c5joN7rUmT2StGPPGA/expected/octopus-1-2.json]: absoluteBounds removed

**Review by cerosnikita — APPROVED** [review]

**Pull request merged**